### DISCUSSION TITLE
Use the first matching version found in a list of repositories

### DISCUSSION THREAD

**Discussion by bilogic** [composer/composer#11354] [Q&A / Support] [answered]
```JSON
"repositories": [
    {
        "type": "path",
        "url": "path/to/package/*/*"
    },
    {
        "type": "composer",
        "url": "https:\/\/example.com"
    }
]
```
> 
>   Problem 1
>   - Root composer.json requires aaa/bbb 1.0.0, it is satisfiable by aaa/bbb[1.0.0] from composer repo (https://example.com) but aaa/bbb[dev-master] from path repo (../../package-repo/packages/*/*) has higher repository priority. The packages from the higher priority repository do not match your constraint and are therefore not installable. That repository is canonical so the lower priority repo's packages are not installable. See https://getcomposer.org/repoprio for details and assistance.
> 
> Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

1. I have the same package in both `repositories` entry, which I will refer to `repo1` (path) and `repo2` (composer)
2. `repo1` contains only `dev-master` versions because it is local
3. `composer` contains released/tagged versions

If I were to require version `1.0.0`, it fails even though it is found in `repo2` because `repo1` is higher priority and has `dev-master`

Is there a way to make composer use the version `1.0.0` found in `repo2` without modifying `composer.json`?

I'm aware this may bring about a change in composer's behavior, but my logic is for composer to use the first matching version it can find going down the list.

**Comment by bilogic** [comment, marked as answer]
Gosh, it's embarassing to find the answer after pasting the answer. This can be fixed by making packages non-canonical.

**Reply by stof** [reply]
note that the link to the doc explaining that is even provided inside the error message. So Composer is already trying to help here
